crates/driver/src/infra/solver/dto/solution.rs

```diff
@@ -204,6 +223,53 @@ impl Solutions {
             })
             .collect()
     }
+
+    // Function to check that a JIT order signature is signed by the solver
+    fn recover_signer_from_jit_trade_order(
+        jit: &JitTrade,
+        signature: &competition::order::Signature,
+        solver: &Solver,
+    ) -> Result<H160, super::Error> {
```

Comment:
Let's return the appropriate type right away.
```suggestion
    ) -> Result<eth::Address, super::Error> {
```